source/DSCResources/DSC_CMNetworkDiscovery/DSC_CMNetworkDiscovery.psm1

```diff
@@ -0,0 +1,155 @@
+$script:dscResourceCommonPath = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\DscResource.Common'
+$script:configMgrResourcehelper = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\ConfigMgrCBDsc.ResourceHelper'
+
+Import-Module -Name $script:dscResourceCommonPath
+Import-Module -Name $script:configMgrResourcehelper
+
+$script:localizedData = Get-LocalizedData -DefaultUICulture 'en-US'
+
+<#
+    .SYNOPSIS
+        This will return a hashtable of results.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER Enabled
+        Specifies the enablement of the Network discovery method.
+        When setting enabled to true, Network Discovery will be enabled with default settings.
+
+        Not used in Get-TargetResource.
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    $networkDiscovery = (Get-CMDiscoveryMethod -Name NetworkDiscovery -SiteCode $SiteCode).Props
+    $enabledStatus = ($networkDiscovery | Where-Object -FilterScript {$_.PropertyName -eq 'Discovery Enabled'}).value1
+
+    if ($enabledStatus -eq 'FALSE')
+    {
+        $enable = $false
+    }
+    else
+    {
+        $enable = $true
+    }
+
+    return @{
+        SiteCode = $SiteCode
+        Enabled  = $enable
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER Enabled
+        Specifies the enablement of the Network discovery method.
+        When setting enabled to true, Network Discovery will be enabled with default settings.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+
+        if ($Enabled -ne $state.Enabled)
```

Comment:
if ($Enabled -ne $state.Enabled)
{
    Write-Verbose -Message ($script:localizedData.SettingEnable -f $state.Enabled, $Enabled)
    
    Set-CMDiscoveryMethod -NetworkDiscovery -SiteCode $SiteCode -Enabled $Enabled
}

Reply:
Tested and changed